src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp

```diff
@@ -0,0 +1,131 @@
+    model = ngraph::builder::subgraph::makeConvPoolRelu();
+}
+
+std::string OVPropertiesTests::getTestCaseName(testing::TestParamInfo<PropertiesParams> obj) {
+    std::string device_name;
+    AnyMap properties;
+    std::tie(device_name, properties) = obj.param;
+    std::ostringstream result;
+    result << "device_name=" << device_name << "_";
+    if (!properties.empty()) {
+        result << "properties=" << util::join(util::split(util::to_string(properties), ' '), "_");
+    }
+    return result.str();
+}
+
+void OVPropertiesTests::SetUp() {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED();
+    std::tie(device_name, properties) = this->GetParam();
+    model = ngraph::builder::subgraph::makeConvPoolRelu();
+}
+
+void OVPropertiesTests::TearDown() {
+    if (!properties.empty()) {
+        utils::PluginCache::get().reset();
+    }
+}
+
+
+TEST_P(OVEmptyPropertiesTests, SetEmptyProperties) {
+    OV_ASSERT_NO_THROW(core->get_property(device_name, ov::supported_properties));
+    OV_ASSERT_NO_THROW(core->set_property(device_name, AnyMap{}));
+}
+
+TEST_P(OVEmptyPropertiesTests, CanCompileModelWithEmptyProperties) {
+    OV_ASSERT_NO_THROW(core->get_property(device_name, ov::supported_properties));
+    OV_ASSERT_NO_THROW(core->compile_model(model, device_name, AnyMap{}));
+}
+
+// Setting correct properties doesn't throw
+TEST_P(OVPropertiesTests, SetCorrectProperties) {
+    OV_ASSERT_NO_THROW(core->get_property(device_name, ov::supported_properties));
+    OV_ASSERT_NO_THROW(core->set_property(device_name, properties));
+}
+
+TEST_P(OVPropertiesTests, CanCompileModelWithCorrectProperties) {
+    OV_ASSERT_NO_THROW(core->compile_model(model, device_name, properties));
+}
+
+TEST_P(OVPropertiesTests, CanUseCache) {
+    core->set_property(ov::cache_dir("./test_cache"));
+    OV_ASSERT_NO_THROW(core->compile_model(model, device_name, properties));
+    OV_ASSERT_NO_THROW(core->compile_model(model, device_name, properties));
+    CommonTestUtils::removeDir("./test_cache");
+}
+
+TEST_P(OVPropertiesTests, canSetPropertyAndCheckGetProperty) {
+    core->set_property(device_name, properties);
+    for (const auto& property_item : properties) {
+        Any property;
+        OV_ASSERT_NO_THROW(property = core->get_property(device_name, property_item.first));
+        ASSERT_FALSE(property.empty());
+        ASSERT_EQ(property, property_item.second);
+    }
+}
+
+TEST_P(OVPropertiesTests, canSetPropertyTwiceAndCheckGetProperty) {
+    core->set_property(device_name, {});
+    core->set_property(device_name, properties);
+    for (const auto& property_item : properties) {
+        Any property;
+        OV_ASSERT_NO_THROW(property = core->get_property(device_name, property_item.first));
+        ASSERT_FALSE(property.empty());
+        ASSERT_EQ(property, property_item.second);
+    }
+}
+
+TEST_P(OVPropertiesIncorrectTests, SetPropertiesWithIncorrectKey) {
+    OV_ASSERT_NO_THROW(core->get_property(device_name, ov::supported_properties));
```

Comment:
why do we need `core->get_property(device_name, ov::supported_properties` in all tests?

Reply:
may be can be removed. Just a copy past.